src/components/ha-base-time-input.ts

```diff
@@ -140,11 +136,11 @@ export class HaBaseTimeInput extends LitElement {
                 id="day"
                 type="number"
                 inputmode="numeric"
-                .value=${this.days}
+                .value=${this.days.toFixed()}
                 .label=${this.dayLabel}
                 name="days"
                 @input=${this._valueChanged}
```

Comment:
Does this really need to be on the `input` event?  Seems kind of superfluous to fire a change to the backend every time a digit is added or deleted.  I would suggest changing to the `change` event.  Any objections?

Reply:
It could lead to timing issues to change it, it would need some testing (esp. in dialogs and stuff)